### DISCUSSION TITLE
Azure Functions

### DISCUSSION THREAD

**Discussion by lcarlso2** [elsa-workflows/elsa-core#4707] [Design & tech]
Considering implementing Elsa using Azure functions as the backend. I know some features won't be possible, but I was just curious on your thoughts.

**Comment by MCovert-SMP** [comment]
Did you ever get Elsa working using Azure Functions as the backend?

**Comment by sfmskywalker** [comment]
Short answer: I don't see an official Azure-Functions-specific backend integration for Elsa today.

What I could confirm:

- There is still an open tracking issue in `elsa-core` for Azure Functions integration: `#3583`.
- There is also an open implementation/integration issue in `elsa-extensions`: `elsa-extensions#39`.
- I do not see a shipped Azure Functions module or sample in the current Elsa repos.

So if by “use Azure Functions as the backend” you mean a first-class Elsa integration, that does not appear to exist yet.

Today the realistic options seem to be:

- host Elsa in a normal ASP.NET Core process and call Azure Functions through HTTP activities, or
- build a custom integration/module on top of HTTP / activity-provider patterns.

If the goal is to run Elsa itself *inside* Azure Functions, I would treat that as unsupported territory at the moment. The current runtime is built around a long-running host with background dispatching, bookmarks, scheduling, and Studio/API hosting, which is not a natural fit for a Functions host.
